src/Concordium/Client/Runner.hs

```diff
@@ -627,7 +664,14 @@ processTransactionCmd action baseCfgDir verbose backend =
       withClient backend $ do
         transferData <- getEncryptedAmountTransferData (naAddr senderAddr) receiverAcc amount index secretKey
 
-        let payload = Types.encodePayload $ Types.EncryptedAmountTransfer (naAddr receiverAcc) transferData
+        let plDo = do
+                res <- case maybeMemo of 
+                  Nothing -> return $ Types.EncryptedAmountTransfer (naAddr receiverAcc) transferData
+                  Just memo -> do
+                    bs <- readMemoAsBytestring memo backend
+                    return $ Types.EncryptedAmountTransferWithMemo (naAddr receiverAcc) (Types.Memo $ BSS.toShort bs) transferData
+                return $ Types.encodePayload res
+        payload <- liftIO plDo
```

Comment:
If you move `readMemoAsBytes` to `ClientMonad` then you don't need this `liftIO`.